### PR TITLE
EZP-30202: Error to preview / publish content item with empty Image Asset field

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30202](https://jira.ez.no/browse/EZP-30202)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.x`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/MVC/Symfony/FieldType/ImageAsset/ParameterProvider.php [hunk 1]
```diff
@@ -26,39 +26,50 @@ class ParameterProvider implements ParameterProviderInterface
...
         try {
             $contentInfo = $this->loadContentInfo(
                 $field->value->destinationContentId
             );
 
-            if (!$this->userHasPermissions($contentInfo)) {
+            if ($this->userHasPermissions($contentInfo)) {
                 return [
-                    'available' => false,
+                    'available' => !$contentInfo->isTrashed(),
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/FieldType/ImageAsset/ParameterProvider.php]: How about:
```php
return [	
    'available' => !$contentInfo->isTrashed() && $this->userHasPermissions($contentInfo),	
];
```
Then you wouldn't need to make empty try-catch.

**Comment by adamwojs** [comment]
> How exactly the fix is related to the issue described in JIRA?

1. In the default template you call `ez_render_field` on Image Asset Field with is empty: https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Bundle/EzPublishCoreBundle/Resources/views/default/content/full.html.twig#L6
2. Then no matter if the field value is empty or not the ParameterProvider for field type is called:
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/FieldRenderingExtension.php#L160-L164 (IMHO this is correct behaviour)
3. Parameter provider call 
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/MVC/Symfony/FieldType/ImageAsset/ParameterProvider.php#L44-L46 with `null` but method expect `int` and the expection is thrown. 

Originaly, `\eZ\Publish\API\Repository\ContentService::loadContentInfo` was always called and described case was handled by try/catch statement, but it not optimal: we know that field value is empty (so we can avoid repository. Ref. https://github.com/ezsystems/ezpublish-kernel/pull/2403/files#diff-a69bf20f91185b4b1c044ed1b71d5bceR36 

* In terms of Image Asset empty value mean `\eZ\Publish\Core\FieldType\ImageAsset\Value::$destinationContentId === null`. Ref. https://github.com/adamwojs/ezpublish-kernel/blob/2e063984af17b93406c8b504f7c5e1f2525ec30a/eZ/Publish/Core/FieldType/ImageAsset/Type.php#L126-L136

**Comment by lserwatka** [comment]
@adamwojs tests are failing here

**Comment by adamwojs** [comment]
@lserwatka It's green now. It was some random failure.

**Comment by lserwatka** [comment]
Thank you, could you merge it up?

**Comment by adamwojs** [comment]
Done.

**Pull request merged**